### DISCUSSION TITLE
Color Profile - Calculating Back

### DISCUSSION THREAD

**Discussion by Schrank97** [ImageMagick/ImageMagick#3282] [Help] [answered]
Hello guys,

today, my topic is kind of.. hard for me to solve.

I work in a company that has to use different printing devices. 
On those printing devices, there are different colorprofiles installed. For our best process, it would be necessairy, to delete those "Default"- Profiles from the printer and apply them on the images that are sent to the printer.

Here comes the "difficult" part of the story:
We create an image out of two seperate images. For one, we need to apply a icc- Profile, but not for the other one.
Our idea was to "Calculate the icc-Profile backwards" on the image that needs the icc. That means, the image itself has no profile embedded, but the Colorchannels are calculated that way, that the printer needs to print them. Is it possible somehow to "Calculate a image with an ICC, and not Embed" it ?

It's a difficult question, but i hope, somebody can help me achieving this.

Best regards
Jakob

PS: I'm using the .Net Implementation of ImageMagick in C#, if it's necessairy for you to know :)

**Comment by snibgo** [comment, marked as answer]
> Is it possible somehow to "Calculate a image with an ICC, and not Embed" it ?

Yes. You can convert to the new profile, then remove it with "+profile *". The pixel values will be converted to that new profile, but no profile will be embedded.

> For one, we need to apply a icc- Profile, but not for the other one.

What do you mean by "apply" an ICC profile? It doesn't really mean anything. There are three operations we can do with ICC profiles:

1. Assign a profile to an image. This means embedding the profile, without changing pixel values. This is often needed when we know an image is sRGB (or some other known colorspace) but has no embedded profile.

2. Convert an image from one profile to another profile. This changes pixel values, and embeds the new profile.

3. Remove an embedded profile, with "+profile *" or "-strip".

Note: "+profile *" etc are CLI operations. I don't know the C# equivalents.

**Reply by Schrank97** [reply]
Hello Snibgo,

thank you for your answer! It helped me very much! 

Stil, working with ICC-Profiles and Colorchannels are pretty hard for me to understand. But I'm very new to this topic and will learn further. But like I said, your suggestion works out perfect.

Best regards
Jakob
